OneSignalSDK/onesignal/core/src/main/java/com/onesignal/user/IUserManager.kt

```diff
@@ -138,4 +149,15 @@ interface IUserManager {
      * Return a copy of all local tags from the current user.
      */
     fun getTags(): Map<String, String>
+
+    /**
+     * Add an observer to the user state, allowing the provider to be
+     * notified whenever the user state has changed.
+     */
```

Comment:
Let's add a callout to developers here on the `addObserver` method documentation:

> Important: When using the observer to retrieve the `onesignalId`, check the `externalId` as well to confirm the values are associated with the expected user.